### DISCUSSION TITLE
How can I serialize/deserialize a custom class containing only data to and from postgres

### DISCUSSION THREAD

**Discussion by ghost** [sqlalchemy/sqlalchemy#9979] [Usage Questions] [answered]
Give the following,
```
data[MyClass] = mapped_column(LargeBinary)
```

How can I correctly serialize/deserialize MyClass?

**Comment by CaselIT** [comment]
Hi,

You may be looking for https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.PickleType

**Reply by ghost** [reply]
Thanks. Is there any direct support for python.dataclasses <-> jsonb? And if so, is there doc on how it is supported?

**Reply by CaselIT** [reply]
Not really. You may be able to serialize a dataclasses to json using a custom json serializer, but there isn't a way of getting back a dataclass from the db

Also note that jsonb is not a large binary.

**Reply by ghost** [reply]
In that sense, can you tell me how's the proper way of using database json with sqlalchemy, specifically, what's the python class type I shall use?

**Reply by ghost** [reply]
And also, fundamentally, I'd like to know if there is some hook functions that I can write that instructs Sqlalchemy to serialize and deserialize a certain column.

**Reply by CaselIT** [reply, marked as answer]
> In that sense, can you tell me how's the proper way of using database json with sqlalchemy, specifically, what's the python class type I shall use?

there isn't any python class connected to a json type in the default type map, since json can contain many type of objects, so you will have to either define a type map yourself or specify the type in the mapped columns.

> And also, fundamentally, I'd like to know if there is some hook functions that I can write that instructs Sqlalchemy to serialize and deserialize a certain column.

You can probably do something similar to what the pickle type does https://github.com/sqlalchemy/sqlalchemy/blob/fdbbc54a95d40c9a72e41b0d9dce010e89b83b8b/lib/sqlalchemy/sql/sqltypes.py#L1773-L1869

There may also be external packages that implements something like this for dataclasses

if you limit yourself to only the orm then you may be able to use attribute events https://docs.sqlalchemy.org/en/20/orm/events.html#attribute-events
